### PR TITLE
QgsGeometry: Add `fromPoint` and `fromBox3D` constructors

### PR DESCRIPTION
**Pull request by ptitjano**
## Description

This adds two new methods to `QgsGeometry`:
- `fromPoint` to generate a `QgsGeometry` from a `QgsPoint`
- `fromBox3D` to generate a `QgsGeometry` from a `QgsBox3D`

cc @lbartoletti @benoitdm-oslandia

### CODE REVIEW

File: tests/src/core/geometry/testqgsgeometry.cpp [hunk 1]
```diff
@@ -744,6 +746,48 @@ void TestQgsGeometry::splitCurve()
   QCOMPARE( p2->asWkt(), curve2 );
 }
 
+void TestQgsGeometry::fromBox3d()
+{
+  QgsBox3D box3d( 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 );
+  QgsGeometry result( QgsGeometry::fromBox3D( box3d ) );
+  QCOMPARE( result.wkbType(), Qgis::WkbType::MultiPolygonZ );
```
Comment (nyalldawson) [inline tests/src/core/geometry/testqgsgeometry.cpp:753]: The best way to do this check is to compare the obtained geometry's wkt against the expected wkt. That makes for a much more descriptive test and one which is easier to debug if things go wrong

**Review by benoitdm-oslandia — APPROVED** [review]
:+1:

File: src/core/geometry/qgsgeometry.cpp [hunk 2]
```diff
@@ -171,6 +171,16 @@ QgsGeometry QgsGeometry::fromPointXY( const QgsPointXY &point )
...
 }
 
+QgsGeometry QgsGeometry::fromPoint( const QgsPoint &point )
+{
+  std::unique_ptr< QgsAbstractGeometry > geom( QgsGeometryFactory::fromPoint( point ) );
+  if ( geom )
+  {
+    return QgsGeometry( geom.release() );
+  }
+  return QgsGeometry();
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.cpp:181]:
```suggestion
  return QgsGeometry( point.clone() );
```

and remove the unnecessary QgsGeometryFactory::fromPoint function.

File: tests/src/core/geometry/testqgsgeometry.cpp (hunk 1 above)
Comment (ptitjano) [reply tests/src/core/geometry/testqgsgeometry.cpp:753]: Done.

File: src/core/geometry/qgsgeometry.cpp (hunk 2 above)
Comment (ptitjano) [reply src/core/geometry/qgsgeometry.cpp:181]: Done.

File: tests/src/python/test_qgsgeometry.py [hunk 3]
```diff
@@ -266,6 +274,21 @@ def testFromMultiPolygon(self):
         ])
         self.assertEqual(myMultiPolygon.wkbType(), QgsWkbTypes.MultiPolygon)
 
+    def testFromBox3D(self):
+        myBox3D = QgsGeometry.fromBox3D(QgsBox3D(1, 2, 3, 4, 5, 6))
+        self.assertEqual(myBox3D.wkbType(), QgsWkbTypes.MultiPolygonZ)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsgeometry.py:279]: Can you also compare wkt here please?

Comment (ptitjano) [reply tests/src/python/test_qgsgeometry.py]: Done.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**